### PR TITLE
Add codelab project - binary number converter

### PR DESCRIPTION
**Pull request by paulchen503**
### What this change is about?

The change contains codelab project that converts binary number to decimal.

### How was it tested?

Unit testing.

### CODE REVIEW

File: codelab/shunbao_codelab/MagicNumberConverter.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package shunbao_codelab;
+
+public class MagicNumberConverter {
+
+    public int binaryToDecimal(String binary)
+    {
+        if (binary=="0") {
```
Comment (paulchen503) [inline codelab/shunbao_codelab/MagicNumberConverter.java]: add spaces around the "==".

File: codelab/shunbao_codelab/MagicNumberConverter.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+               return 0;
+        }else if(binary == "1") {
+            return 1;
+        } else if (binary == "10") {
+         return 2;
+        } else if (binary == "11") {
+            return 3;
+        }
+        int s=0 ;
```
Comment (paulchen503) [inline codelab/shunbao_codelab/MagicNumberConverter.java:16]: does not handle input like "1111", please fix it.

File: codelab/shunbao_codelab/MagicNumberConverterTest.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import org.junit.Test;
+
+public class MagicNumberConverterTest {
+    @Test
+    public void testConvertBinaryNumber() {
+        MagicNumberConverter convert = new MagicNumberConverter();
+        int result = convert.binaryToDecimal("0");
+        assertEquals(0, result);
+
+        assertEquals(3, convert.binaryToDecimal("11"));
```
Comment (paulchen503) [inline codelab/shunbao_codelab/MagicNumberConverterTest.java:14]: please also test invalid cases such as empty string, and also test negative number like "-11". Please test long number such as "11001111". 

Testing super long number to see if there is overflow: "11111111111111"

**Review by paulchen503 — COMMENTED** [review]
Please address the feedback before merge.
